test/unit/src/common/components/user-avatar/t_user-avatar.js

```diff
@@ -1,22 +1,41 @@
 import React from 'react';
 import expect from 'expect';
 import { shallow } from 'enzyme';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
 
-import { UserAvatar } from '../../../../../../src/common/components/user-avatar';
+import UserAvatar from '../../../../../../src/common/components/user-avatar';
 import { HeadingOne, HeadingThree } from '../../../../../../src/common/components/vanilla/heading';
 
+const middlewares = [ thunk ];
+const mockStore = configureMockStore(middlewares);
+
 describe('<UserAvatar />', function() {
   let element;
+  let store;
+  let auth;
+  let user;
+
+  beforeEach(() => {
+    store = mockStore({
+      auth,
+      user
+    });
+
+    // shallow render container component with the store
+    element = shallow(<UserAvatar store={store} />)
```

Comment:
Having had a look around and tried it myself, I'm -1 for testing the component with the store, it's essentially a test of redux. If the mapStateToProps function was complex I would export that as a named export along with the component and unit test it too.